reader/reader_test.go

```diff
@@ -0,0 +1,138 @@
+package reader
+
+import (
+	"bytes"
+	"io"
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestRead(t *testing.T) {
+	data := []byte(`Header line
```

Comment:
Could this not be structured like so:
`data := []byte("Header line\nFirst line\nSecond line\n")`?

This would make it neater.

Reply:
The "\n"s have to be in double quotes and then concatenated so its not quite that simple but yes it can be collapsed to one line so I've done that